pkg/dockerregistry/server/prune/restore.go

```diff
@@ -0,0 +1,377 @@
+package prune
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/digest"
+	"github.com/docker/distribution/manifest/schema2"
+	"github.com/docker/distribution/reference"
+	"github.com/docker/distribution/registry/storage/driver"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	dockerapiv10 "github.com/openshift/api/image/docker10"
+	imageapiv1 "github.com/openshift/api/image/v1"
+	"github.com/openshift/image-registry/pkg/dockerregistry/server/client"
+	"github.com/openshift/image-registry/pkg/dockerregistry/server/manifesthandler"
+	regstorage "github.com/openshift/image-registry/pkg/dockerregistry/server/storage"
+	"github.com/openshift/image-registry/pkg/imagestream"
+	imageapi "github.com/openshift/image-registry/pkg/origin-common/image/apis/image"
+	originutil "github.com/openshift/image-registry/pkg/origin-common/util"
+)
+
+// Restore defines a common set of operations for database and storage validation
+type Restore interface {
+	BrokenImage(image imageapiv1.Image, err error) error
+	BrokenImageStreamTag(is imageapiv1.ImageStream, tag imageapiv1.NamedTagEventList, pos int, err error) error
+	ImageStreamTag(imageStream imagestream.ImageStream, image *imageapiv1.Image, tagName string) error
+}
+
+type StorageRestore struct {
+	Ctx    context.Context
+	Client client.Interface
+}
+
+var _ Restore = &StorageRestore{}
+
+func (r *StorageRestore) ImageStreamTag(imageStream imagestream.ImageStream, image *imageapiv1.Image, tagName string) error {
+	return imageStream.CreateImageStreamMapping(r.Ctx, r.Client, tagName, image)
+}
+
+func (r *StorageRestore) BrokenImage(image imageapiv1.Image, err error) error {
+	return nil
+}
+
+func (r *StorageRestore) BrokenImageStreamTag(is imageapiv1.ImageStream, tag imageapiv1.NamedTagEventList, pos int, err error) error {
+	return nil
```

Comment:
same here, shouldn't this at least print something, since it's not fixing it?

Reply:
I don't think so because for this, there is a check mode.